### PR TITLE
Add support to run aggregate operations in parallel

### LINKED ISSUE #642: Aggregation breaks project serialization for parallel execution
### Description
Using `@aggregator()` breaks project serialization. This prevents users from running commands in parallel, like `python project.py run -o my_operation --parallel`. This breaks all operations, even if the operation being run is not an aggregate operation. Discovered by @jdice.

### To reproduce

```python
import cloudpickle
import signac
from flow import FlowProject, aggregator


@aggregator()
@FlowProject.operation
def agg_op(*jobs):
    pass


if __name__ == "__main__":
    pr = signac.init_project("test")
    pr.open_job({"a": 1}).init()

    pr = FlowProject()
    pkl = cloudpickle.dumps(pr)
    cloudpickle.loads(pkl)
```

### Error output

```pytb
Traceback (most recent call last):
  File "/home/bdice/tmp_signac/project.py", line 18, in <module>
    cloudpickle.loads(pkl)
  File "/home/bdice/code/signac/signac/contrib/job.py", line 941, in __setstate__
    self.statepoint._jobs.append(self)
  File "/home/bdice/code/signac/signac/contrib/job.py", line 476, in statepoint
    jobs=[self], filename=self._statepoint_filename
  File "/home/bdice/code/signac/signac/contrib/job.py", line 360, in _statepoint_filename
    return os.sep.join((self.workspace(), self.FN_MANIFEST))
  File "/home/bdice/code/signac/signac/contrib/job.py", line 352, in workspace
    self._wd = os.sep.join((self._project.workspace(), self.id))
  File "/home/bdice/code/signac/signac/contrib/project.py", line 379, in workspace
    if self._wd is None:
AttributeError: 'FlowProject' object has no attribute '_wd'
```

### System configuration

 - Version of Python: 3.9 
 - Version of signac: 1.7.0 (development version)
 - Version of signac-flow: 0.19.0 (development version)
 - Version of cloudpickle: 2.0.0. I also confirmed this is broken in cloudpickle 1.6.0.

### PR DESCRIPTION
**Pull request by kidrahahjo**
## Description
Aggregate operations can now execute in parallel mode.

## Motivation and Context
Fixes #642 

## Checklist:

- [x] I am familiar with the [Contributing Guidelines](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [Contributor Agreement](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.
- [x] The [package documentation](https://github.com/glotzerlab/signac-flow/tree/master/doc) and [framework documentation](https://docs.signac.io/) in [signac-docs](https://github.com/glotzerlab/signac-docs) are up to date with these changes.
- [x] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt) and added any related issue and pull request numbers for future reference.

### CODE REVIEW

**Comment by kidrahahjo** [comment]
Too many dependent tests 😪, I'll have to test it locally.

**Review by bdice — APPROVED** [review]
Really nice fix @kidrahahjo! It turns out to be simpler than I had anticipated. The test also looks great. I pushed a couple small changes. This is good to go once it receives a second review.

**Comment by bdice** [comment]
CI was failing for our "oldest" requirements. The cause was an inability to serialize `dict_values` objects in older versions of cloudpickle. I have updated our pinning from `cloudpickle>=1.1.1` to `cloudpickle>=1.6.0`. Version 1.6.0 was released on August 25, 2020 and the current version is 2.1.0.

**Review by bdice — APPROVED** [review]

**Pull request merged**
